azure-mgmt-search/src/main/java/com/microsoft/azure/management/search/implementation/QueryKeyImpl.java

```diff
@@ -0,0 +1,40 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for
+ * license information.
+ */
+package com.microsoft.azure.management.search.implementation;
+
+import com.microsoft.azure.management.resources.fluentcore.model.implementation.WrapperImpl;
+import com.microsoft.azure.management.search.QueryKey;
+
+/**
+ * Describes an API key for a given Azure Search service that has permissions for query operations only.
+ */
+class QueryKeyImpl extends WrapperImpl<QueryKeyInner> implements QueryKey {
+  /**
+   * Create QueryKey.
+   */
+  protected QueryKeyImpl(QueryKeyInner innerObject) {
+    super(innerObject);
+  }
+
+  /**
+   * Get the name value.
```

Comment:
if this is our internal IMpl then 
a) it should have `@Override` where appropriate
a) it shouldn't have javadocs on overrides